sdk/core/azure-core/src/http/telemetry_policy.cpp

```diff
@@ -17,13 +16,17 @@
 #endif
 
 #include <windows.h>
+#elif defined(AZ_PLATFORM_POSIX)
+#include <sys/utsname.h>
+#endif
 
 namespace {
-
 std::string GetOSVersion()
 {
   std::ostringstream osVersionInfo;
 
+#if defined(AZ_PLATFORM_WINDOWS)
+#if !defined(WINAPI_PARTITION_DESKTOP) || WINAPI_PARTITION_DESKTOP
```

Comment:
Wait I am confused, isn't this `if !A || A` which is always true?

Reply:
Good catch :)

Reply:
No, no! What are you talking about? :)
It IS correct.
It says "`if IsDesktop is not DEFINED or if IsDesktop()`". Two different things.
Basically, if you compile on where `WINAPI_PARTITION_DESKTOP` is not defined, it is a pre-UWP-era Windows machine, compile as windows Desktop. Otherwise, if you compile on some Win10 machine, with an SDK that does know about UWP and non-UWP being out there, then ask what `WINAPI_PARTITION_DESKTOP` is telling you. It is supposed to evaluate to true if building for good old Win32 as we know it, and false otherwise.